apps/hermes/client/js/src/HermesClient.ts

```diff
@@ -0,0 +1,228 @@
+   * @param config Optional HermesClientConfig for custom configurations.
+   */
+  constructor(endpoint: string, config?: HermesClientConfig) {
+    this.baseURL = endpoint;
+    this.timeout = config?.timeout ?? DEFAULT_TIMEOUT;
+    this.httpRetries = config?.httpRetries ?? DEFAULT_HTTP_RETRIES;
+  }
+
+  private async httpRequest<ResponseData>(
+    url: string,
+    schema: z.ZodSchema<ResponseData>,
+    options?: RequestInit,
+    retries = this.httpRetries,
+    backoff = 100 + Math.floor(Math.random() * 100), // Adding randomness to the initial backoff to avoid "thundering herd" scenario where a lot of clients that get kicked off all at the same time (say some script or something) and fail to connect all retry at exactly the same time too
+    externalAbortController?: AbortController
+  ): Promise<ResponseData> {
+    const controller = externalAbortController || new AbortController();
+    const { signal } = controller;
+    options = { ...options, signal }; // Merge any existing options with the signal
+
+    // Set a timeout to abort the request if it takes too long
+    const timeout = setTimeout(() => controller.abort(), this.timeout);
+
+    try {
+      const response = await fetch(url, options);
+      clearTimeout(timeout); // Clear the timeout if the request completes in time
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+      }
+      const data = await response.json();
+      return schema.parse(data);
+    } catch (error) {
+      clearTimeout(timeout);
+      if (
+        retries > 0 &&
+        !(error instanceof Error && error.name === "AbortError")
+      ) {
+        // Wait for a backoff period before retrying
+        await new Promise((resolve) => setTimeout(resolve, backoff));
+        return this.httpRequest(url, schema, options, retries - 1, backoff * 2); // Exponential backoff
+      }
+      throw error;
+    }
+  }
+
+  /**
+   * Fetch the set of available price feeds.
+   * This endpoint can be filtered by asset type and query string.
+   * This will throw an error if there is a network problem or the price service returns a non-ok response.
+   *
+   * @param options Optional parameters:
+   *        - query: String to filter the price feeds. If provided, the results will be filtered to all price feeds whose symbol contains the query string. Query string is case insensitive. Example: "bitcoin".
+   *        - filter: String to filter the price feeds by asset type. Possible values are "crypto", "equity", "fx", "metal", "rates". Filter string is case insensitive.
+   *
+   * @returns Array of PriceFeedMetadata objects.
+   */
+  async getPriceFeeds(options?: {
+    query?: string;
+    filter?: string;
+  }): Promise<PriceFeedMetadata[]> {
+    const url = new URL("/v2/price_feeds", this.baseURL);
+    if (options) {
+      this.appendUrlSearchParams(url, options);
+    }
+    return await this.httpRequest<PriceFeedMetadata[]>(
```

Comment:
You shouldn't have to pass type params to the calls to `httpRequest` any more now that you have schemas and type inference -- and in fact if removing the type params fails then it means something isn't working correctly with inference so I'd suggest not shipping it